File: Python/ordenacion_insercion.py
import random

def insertion_sort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1
        while j >= 0 and key < arr[j]:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key

# Generar lista de números aleatorios
arr = [random.randint(1, 100) for _ in range(10)]
print("Antes de ordenar:", arr)
insertion_sort(arr)
print("Después de ordenar:", arr)

File: JavaScript/ordenacion_insercion.js
function insertionSort(arr) {
    for (let i = 1; i < arr.length; i++) {
        let key = arr[i];
        let j = i - 1;
        while (j >= 0 && arr[j] > key) {
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = key;
    }
}

// Generar array aleatorio
let arr = Array.from({length: 10}, () => Math.floor(Math.random() * 100) + 1);

console.log("Antes de ordenar:", arr);
insertionSort(arr);
console.log("Después de ordenar:", arr);